docs/site/src/plugins/inject-code/injectLoader.js

```diff
@@ -77,12 +88,23 @@ const addCodeInject = function (source) {
               const funName = getName(marker, funKey);
               const structName = getName(marker, structKey);
               const moduleName = getName(marker, moduleKey);
+              const variableName = getName(marker, varKey);
+              const useName = getName(marker, useKey);
+              const componentName = getName(marker, componentKey);
+              const enumName = getName(marker, enumKey);
+              const typeName = getName(marker, typeKey);
               if (funName) {
                 const funs = funName.split(",");
                 let funContent = [];
+
                 for (let fn of funs) {
                   fn = fn.trim();
-                  const funStr = `^(\\s*)*?(public )?fun \\b${fn}\\b.*?}\\n(\\s*?})?(?=\\n)?`;
+                  let funStr = "";
+                  if (isMove) {
+                    funStr = `^(\\s*)*?(pub(lic)? )?fu?n \\b${fn}\\b.*?}\\n(\\s*?})?(?=\\n)?`;
+                  } else if (isTs) {
+                    funStr = `^( *)(async )?(export (default )?)?function \\b${fn}\\b.*?\\n\\1}\\n`;
```

Comment:
Probably should allow arbitrary leading whitespace for TS because we do use tab indentation there.

```suggestion
                    funStr = `^(\\s*)(async )?(export (default )?)?function \\b${fn}\\b.*?\\n\\1}\\n`;
```